src/playlist.js

```diff
@@ -374,30 +375,54 @@ export const seekable = function(playlist, expired, liveEdgePadding) {
  * @param {number} startTime
  * @return {Object}
  */
-export const getMediaInfoForTime = function(
+export const getMediaInfoForTime = function({
   playlist,
   currentTime,
-  startIndex,
+  segmentIndex,
+  partIndex,
   startTime
-) {
+}) {
 
-  const partsAndSegments = getPartsAndSegments(playlist);
   let time = currentTime - startTime;
+  const partsAndSegments = getPartsAndSegments(playlist);
+
+  let startIndex = 0;
+
+  for (let i = 0; i < partsAndSegments.length; i++) {
+    const partAndSegment = partsAndSegments[i];
+
+    if (segmentIndex !== partAndSegment.segmentIndex) {
+      continue;
+    }
+
+    // skip this if part index does not match.
+    if (typeof partIndex === 'number' && typeof partAndSegment.partIndex === 'number' && partIndex !== partAndSegment.partIndex) {
```

Comment:
question: why would `partIndex` not be a number? Is it if it isn't present?

Reply:
I guess below I can see that it is default to `null`. Maybe we should default it to `-1` instead, that way we would only need to check `partIndex !== partAndSegment.partIndex`?

Reply:
We ran into problems with mediaIndex being negative previously during live playlist updates. I wanted to avoid the whole issue with partIndex. The issue was something like: If we get a playlist update at just the right time we can end up downloading the same segment again because `-1 + 1 = 0`. Using null we prevent choosing an incorrect segment via incrementing.